Update helm chart to accomodate recent project changes

ShaunMaher opened the pull request:
* Remove the `--run_gunicorn ` startup flag
* Tidy up some helm chart build edge cases (i.e. can't increment version if there is no previous version)
* Update build process to use the new chart name (`-helm` suffix removed).  Thanks to @Lunik for forcing me to reevaluate my previous assumption about the helm chart name conflicting with the image name.  It seems that it "just works".
* Integrate helm chart build process into the process that builds the docker images so that this becomes a single step (#2040)
* Tweak the GitHub workflow that builds the docker images so it only tries to push to docker hub if the project running the build is `BerriAI/litellm`.  This allows other developers to fork this repository and test the workflows on their forks, without them always failing because they (obviously) can't push to docker hub as BerriAI.
* This should also close the deployment parts of #1880, but mainly because @Lunik's work on #2018

ShaunMaher committed:
Copy helm chart build and deploy job from ghcr_helm_deploy to ghcr_deploy.

ShaunMaher committed:
Remove --run_gunicorn from proxy command line arguments

ShaunMaher committed:
Test: Can we have an OCI image and Helm chart with the same name but different tags?

ShaunMaher committed:
Merge branch 'update_helm_chart_deployment_2024-02-22' of github.com:ShaunMaher/litellm into update_helm_chart_deployment_2024-02-22

ShaunMaher committed:
Don't fail if there is no existing helm chart to get the version from

ShaunMaher committed:
Don't fail if there is no existing helm chart to get the version from

ShaunMaher committed:
No existing chart causes failure

ShaunMaher committed:
Can't use tags, only semvers.

ShaunMaher committed:
Merge branch 'BerriAI:main' into update_helm_chart_deployment_2024-02-22

ishaan-jaff commented:
cc@Lunik can we get your feedback on this too - since you use this ?

dependabot[bot] committed:
build(deps): bump orjson from 3.9.7 to 3.9.15

Bumps [orjson](https://github.com/ijl/orjson) from 3.9.7 to 3.9.15.
- [Release notes](https://github.com/ijl/orjson/releases)
- [Changelog](https://github.com/ijl/orjson/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ijl/orjson/compare/3.9.7...3.9.15)

---
updated-dependencies:
- dependency-name: orjson
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
fix(proxy_server.py): support cost tracking if general_settings is none

works if database_url is in env

debdutdeb commented:
Would be amazing to have this merged  🙏🏼  I'd rather not copy the whole stuff to another chart to use as dependency. 

Please let me know if I can help. 

From the looks of it, structurally it looks all good, but I've not looked deep in.

ishaan-jaff commented:
@debdutdeb it would help us if you could check if this helm chart works for you ?

ShaunMaher committed:
Merge branch 'BerriAI:main' into update_helm_chart_deployment_2024-02-22

Commit:
docs(virtual_keys.md): cleanup doc

ishaan-jaff commented on deploy/charts/litellm/templates/deployment.yaml:
```diff
@@ -161,7 +161,6 @@ spec:
           args:
             - --config
             - /etc/litellm/config.yaml
-            - --run_gunicorn
```
why remove this ?

ishaan-jaff reviewed (changes requested):
re `run_gunicorn`

Commit:
fix: add dependency for google-cloud-aiplatform

ishaan-jaff committed:
(feat) support azure/gpt-instruct models

ishaan-jaff committed:
(fix) support streaming for azure/instruct models

Commit:
fix(anthropic.py): support streaming with function calling

ishaan-jaff committed:
(docs) using azure_text models

Commit:
fix(anthropic.py): fix streaming

Commit:
fix(anthropic.py): fix linting errors

ishaan-jaff committed:
(feat) v0 support command-r

ishaan-jaff committed:
Merge pull request #2471 from BerriAI/litellm_azure_instruct

(feat) support azure/gpt-instruct models

ishaan-jaff committed:
(feat) cohere_chat provider

Commit:
fix(openai.py): return model name with custom llm provider for openai compatible endpoints

ishaan-jaff committed:
(test) command_r

ishaan-jaff committed:
(docs) cohere-comand-r

ishaan-jaff committed:
(feat) exception mapping for cohere_chat

ishaan-jaff committed:
(feat) add cohere_chat to model_prices

ishaan-jaff committed:
(feat) use model json to get cohere_models

Commit:
fix(azure.py): support cost tracking for azure/dall-e-3

Commit:
fix(proxy_server.py): fix /user/info for non-existent user id

Commit:
fix(main.py): trigger new build

Commit:
bump: version 1.31.3 → 1.31.4

ishaan-jaff committed:
Merge pull request #2474 from BerriAI/litellm_support_command_r

[New-Model] Cohere/command-r

Commit:
test: add more logging for failing test

ishaan-jaff committed:
(fix) patch dynamoDB tea_model_alias bug

ishaan-jaff committed:
Merge pull request #2478 from BerriAI/litellm_fix_dynamo_team_alias

(fix) patch dynamoDB team_model_alias bug

ishaan-jaff committed:
(v0) tool calling

ishaan-jaff committed:
(fix) failing cohere test

Commit:
test: set verbose for test

Commit:
Merge pull request #2475 from BerriAI/litellm_azure_dall_e_3_cost_tracking

fix(azure.py): support cost tracking for azure/dall-e-3

Commit:
Merge pull request #2473 from BerriAI/litellm_fix_compatible_provider_model_name

fix(openai.py): return model name with custom llm provider for openai-compatible endpoints (e.g. mistral, together ai, etc.)

ishaan-jaff committed:
(feat) cohere tool calling

ishaan-jaff committed:
(feat) support tool_calling on cohere command-r

ishaan-jaff committed:
(fix) use cohere_chat optional params

Commit:
fix(proxy_server.py): cache master key check

ishaan-jaff committed:
(fix) default langfuse setting

ishaan-jaff committed:
(fix) locust load test

Commit:
Merge pull request #2483 from BerriAI/litellm_proxy_perf

fix(proxy_server.py): cache master key check

Commit:
test(anthropic.py): more logging for test

Commit:
Merge pull request #2382 from BerriAI/litellm_cost_tracking_fix

fix(proxy_server.py): support cost tracking if general_settings is none

Commit:
bump: version 1.31.4 → 1.31.5

Commit:
fix(anthropic.py): concurrent request fix

Commit:
fix(anthropic.py): bug fix

ishaan-jaff committed:
Merge pull request #2484 from BerriAI/litellm_fix_langfuse_default

[Proxy-High Traffic] Fix using Langfuse in high traffic

Commit:
fix(anthropic.py): add more logging

ishaan-jaff committed:
Merge pull request #2479 from BerriAI/litellm_cohere_tool_call

[FEAT Cohere/command-r tool calling

Commit:
docs(cost_tracking.md): add docs for cost tracking dall e 3 calls

Commit:
Merge pull request #2472 from BerriAI/litellm_anthropic_streaming_tool_calling

fix(anthropic.py): support claude-3 streaming with function calling

Commit:
bump: version 1.31.5 → 1.31.6

Commit:
docs(sidebar.js): add dall e 3 cost tracking to docs

ishaan-jaff committed:
(fix) raising No healthy deployment

ishaan-jaff committed:
(fix) return 429 error

ishaan-jaff committed:
(fix) errors fro litellm proxy

ishaan-jaff committed:
Merge pull request #2470 from tylerbrandt/gcp-dep

fix: add dependency for google-cloud-aiplatform

ishaan-jaff committed:
Merge pull request #2204 from BerriAI/dependabot/pip/orjson-3.9.15

build(deps): bump orjson from 3.9.7 to 3.9.15

ishaan-jaff committed:
(fix) issue with using litellm enterprise license

Commit:
feat(prompt_injection_detection.py): support simple heuristic similarity check for prompt injection attacks

Commit:
fix(factory.py): fix mistral api prompt formatting

Commit:
bump: version 1.31.6 → 1.31.7

ishaan-jaff committed:
Merge pull request #2500 from BerriAI/litellm_fix_using_enterpise

(fix) issue with using litellm enterprise on Admin UI

ishaan-jaff committed:
bump: version 1.31.7 → 1.31.8

Commit:
fix(proxy_server.py): support checking openai user param

ishaan-jaff committed:
(fix) using enterprise folder on litellm

Commit:
docs: refactor team based logging in docs

Commit:
Merge pull request #2498 from BerriAI/litellm_prompt_injection_detection

feat(prompt_injection_detection.py): support simple heuristic similarity check for prompt injection attacks

Commit:
docs(enterprise.md): add prompt injection detection to docs

Commit:
bump: version 1.31.8 → 1.31.9

ishaan-jaff committed:
Merge pull request #2493 from BerriAI/litellm_return_429_no_models_available

[Proxy] return 429 when no models available

ishaan-jaff committed:
(test) using get_predict_spend_logs

ishaan-jaff committed:
(fix) predict/spend/logs test

ishaan-jaff committed:
Merge pull request #2501 from BerriAI/litellm_fix_using_enterprise_docker

(fix) using enterprise folder on litellm Docker

ishaan-jaff committed:
(ci/cd) use fake-openai-endpoint in docker builds

ishaan-jaff committed:
Create load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Create locustfile.py

ishaan-jaff committed:
(fix) example fake openai endpoint

ishaan-jaff committed:
(fix) fake openai endpoint

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
(fix) default dockerfile use num_workers = 1

ishaan-jaff committed:
Update load_test.yml

bachya committed:
Add support for `claude-3-haiku-20240307` (`anthropic`)

ishaan-jaff committed:
Update load_test.yml

Commit:
fix(proxy_server.py): move to using UPDATE + SET for track_cost_callback

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

bachya committed:
Update docs

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Merge pull request #2505 from bachya/claude-3-haiku

Add support for `claude-3-haiku-20240307` (`anthropic`)

Commit:
fix(proxy_server.py): maintain support for model specific budgets

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
Update load_test.yml

ishaan-jaff committed:
(fix) update load test result

ishaan-jaff committed:
(ci/cd) upload load test result

ishaan-jaff committed:
(fix) results_stats

ishaan-jaff committed:
(fix) load test action

ishaan-jaff committed:
(fix) update github release

ishaan-jaff committed:
(fix) load test action

ishaan-jaff committed:
(fix) load test action

ishaan-jaff committed:
(fix) upload load_test.html

ishaan-jaff committed:
(fix) load test

ishaan-jaff committed:
(fix) run load test

ishaan-jaff committed:
(fix) run load test

ishaan-jaff committed:
(fix) load test

Commit:
fix(proxy_server.py): fix key caching logic

ishaan-jaff committed:
(feat) add claude-3-haiku

ishaan-jaff committed:
Merge pull request #2508 from BerriAI/litellm_add_bedrock_claude_haiku

(feat) add claude-3-haiku

Commit:
Merge pull request #2506 from BerriAI/litellm_update_db_perf_improvements

fix(proxy_server.py): move to using UPDATE + SET for track_cost_callback

ishaan-jaff committed:
(fix) importing PromptInjectionDetection

ishaan-jaff committed:
bump: version 1.31.9 → 1.31.10

Commit:
Update README.md

ishaan-jaff committed:
(ci/cd) run testing again

ishaan-jaff committed:
(fix-ci-cd) skip deep infra 429 errors

ishaan-jaff committed:
(fix) run prisma generate in default dockerfile

Commit:
fix(proxy_server.py): improve error message on ui login error

ishaan-jaff committed:
Merge pull request #2519 from BerriAI/litellm_run_prisma_generate_default_dockerfile

(fix) run prisma generate in default dockerfile

Commit:
fix(parallel_request_limiter.py): handle metadata being none

Commit:
refactor(main.py): trigger new build

Commit:
fix(utils.py): move to using `litellm.modify_params` to enable max output token trimming fix

Commit:
fix(caching.py): fix print statements

Commit:
refactor(main.py): trigger new build

Commit:
bump: version 1.31.10 → 1.31.11

Commit:
fix(caching.py): support redis caching with namespaces

Commit:
bump: version 1.31.11 → 1.31.12

Commit:
docs(caching.md): add redis namespaces to docs

Commit:
fix(router.py): add no-proxy support for router

Commit:
docs(bedrock.md): adding docs for calling bedrock models on proxy via config.yaml

Commit:
build(model_prices_and_context_window.json): add mistral and mixtral bedrock pricing

ishaan-jaff committed:
(docs) load testing proxy

ishaan-jaff committed:
(temp) test build without using argon2

ishaan-jaff committed:
(fix) load test run over 300s

ishaan-jaff committed:
(fix) - update user error

ishaan-jaff committed:
(fix) multiplatform docker db builds

Commit:
Merge pull request #2520 from BerriAI/litellm_max_tokens_fix

fix(utils.py): move to using `litellm.modify_params` to enable max output token trimming fix

ishaan-jaff committed:
Merge pull request #2525 from BerriAI/litellm_multi_platform_docker_db

(fix) multiplatform docker db builds

Commit:
bump: version 1.31.12 → 1.31.13

ishaan-jaff committed:
(fix) prod load test

ishaan-jaff committed:
(fix) locust load test

ishaan-jaff committed:
(fix) posting release notes on discord action

Commit:
Add function call result submission support for Claude 3 models

ishaan-jaff committed:
(ci/cd) fix litellm triggers on commits

snekkenull committed:
Add groq/gemma-7b-it

snekkenull committed:
Merge branch 'BerriAI:main' into main

ishaan-jaff committed:
(ci/cd) check actions run

ishaan-jaff committed:
(docs) how to run a locust load test

ishaan-jaff committed:
Merge pull request #2529 from snekkenull/main

(feat) add groq/gemma-7b-it

Commit:
Refactor  to allow tool invoke conversion and tool results submission; Allow alternating consecutuve message roles and fix a bug

Commit:
fix(factory.py): raise exception on invalid message being received

ishaan-jaff committed:
(docs) load test proxy

ishaan-jaff committed:
(fix) error cli users see when importing enterprise folder

ishaan-jaff committed:
bump 1.31.13 -> 1.31.14

ishaan-jaff committed:
(ci/cd) check linked triggers

ishaan-jaff committed:
(ci/cd) check triggers

Hitro147 committed:
Exclude /docs to reduce Docker image size

ishaan-jaff committed:
Merge pull request #2524 from BerriAI/litellm_fix_update_user

(fix) - update user error

Hitro147 committed:
Update .dockerignore to exclude additional directories

Commit:
feat(utils.py): add native fireworks ai support

 addresses - https://github.com/BerriAI/litellm/issues/777, https://github.com/BerriAI/litellm/issues/2486

Commit:
docs(fireworks_ai.md): add fireworks ai to docs

Commit:
fix(bedrock.py): add all supported bedrock / anthropic messages api params

ishaan-jaff committed:
Merge pull request #2534 from Hitro147/improve-docker-image-size

Exclude /docs to reduce Docker image size

Commit:
refactor(main.py): trigger new build

Commit:
bump: version 1.31.14 → 1.31.15

Commit:
docs(audio_transcription.md): add openai sdk usage example to audio transcription docs

Commit:
fix(cohere.py): return usage as a pydantic object not dict

Commit:
Merge pull request #2535 from BerriAI/litellm_fireworks_ai_support

feat(utils.py): add native fireworks ai support

Commit:
docs(cohere.md): fix model name in cohere docs

Commit:
Add tool result submission to claude 3 function call test and claude 3 multi-turn conversion to ensure alternating message roles

ishaan-jaff committed:
(ci/cd) interpret load tests

ishaan-jaff committed:
(ci/cd) interpret load test results

ishaan-jaff committed:
(ci/cd) interpret results

ishaan-jaff committed:
(ci/cd) interpret load test results

ishaan-jaff committed:
(ci/cd) interpret_load_test.py

ishaan-jaff committed:
(ci/cd) show load test output on release

ishaan-jaff committed:
(ci/cd) update with load test results

ishaan-jaff committed:
(ci/cd) show load test info on release

ishaan-jaff committed:
(ci/cd) view load test results

ishaan-jaff committed:
(ci/cd) load test results

ishaan-jaff committed:
(ci/cd) show load test results on release

ishaan-jaff committed:
(ci/cd) update release body

ishaan-jaff committed:
(ci/cd) visualize load test results

ishaan-jaff committed:
(ci/cd) show load test results

ishaan-jaff committed:
(ci/cd) view load test results

ishaan-jaff committed:
(ci/cd) view load test results

ishaan-jaff committed:
(ci/cd) run load test

ishaan-jaff committed:
(fix) locustfile used in load test

ishaan-jaff committed:
(ci/cd) use deployed github release for load testing

ishaan-jaff committed:
(ci/cd) load test should allow only 5% fails

Commit:
feat(batch_redis_get.py): batch redis GET requests for a given key + call type

reduces the number of GET requests we're making in high-throughput scenarios

ishaan-jaff committed:
(fix) fix error message raised on passing invalid tokens

udit-001 committed:
docs(langfuse): add chatlitellm section

Commit:
feat(batch_redis_get.py): batch redis GET requests for a given key + call type

reduces number of redis requests. 85ms latency improvement over 3 minutes of load (19k requests).

ishaan-jaff committed:
(fix) run prod load tests for 5 mins

ishaan-jaff committed:
(fix) run load test on new releases, workflow dispatches

udit-001 committed:
docs(langfuse): update chatlitellm section

udit-001 committed:
docs(langfuse): fix missing litellm import

udit-001 committed:
docs(langfuse): update section titles

udit-001 committed:
docs(langfuse): update langfuse casing

ishaan-jaff committed:
(ci/cd) load test

ishaan-jaff committed:
(ci/cd) run load test

ishaan-jaff committed:
(ci/cd) load test

ishaan-jaff committed:
(ci/cd) final working load test / benchmark

ishaan-jaff committed:
Merge pull request #2540 from BerriAI/litellm_fix_error_message_on_bad_key

(fix) Proxy - fix error message raised on passing invalid tokens

ishaan-jaff committed:
(fix) docs litellm helm chart

ishaan-jaff committed:
(docs) using litellm helm

ishaan-jaff committed:
(docs) using litellm + helm charts

ishaan-jaff committed:
(fix) clean up litellm helm chart

ishaan-jaff committed:
(ci/cd) dispatch to run load test

ishaan-jaff committed:
(ci/cd) dispatch to run load test

ishaan-jaff committed:
Merge pull request #2544 from BerriAI/litellm_helm_chart_fixes

[Docs+Fixes] Litellm helm chart use k8 1.21

ishaan-jaff committed:
Merge pull request #2541 from udit-001/docs/chatlitellm-langfuse

docs(langfuse): add chatlitellm section

ishaan-jaff committed:
(docs) litellm + helm chart

Commit:
test(test_caching.py): fix async tests

Commit:
test(test_custom_logger.py): fix async streaming test

Commit:
fix(utils.py): async add to cache - for streaming

Commit:
Merge pull request #2542 from BerriAI/litellm_redis_perf_improvements

89% Caching improvement - Async Redis completion calls + batch redis GET requests for a given key + call type

Commit:
bump: version 1.31.15 → 1.31.16

Commit:
Update proxy_server_config.yaml

Commit:
Update Dockerfile

ishaan-jaff committed:
(fix) errant print saying "Error importing .." msg

Commit:
Update proxy_server_config.yaml

Commit:
Update proxy_server_config.yaml

Commit:
docs(caching.md): add batch redis requests to docs

ishaan-jaff committed:
(fix) backoff import

ishaan-jaff commented:
working on getting these changes merged now !

ishaan-jaff committed:
(feat) publish litellm helm charts

Commit:
Merge branch 'main' into support_anthropic_function_result

Commit:
Merge pull request #2527 from lazyhope/support_anthropic_function_result

Add function call result submission support for Claude 3 models

ishaan-jaff committed:
(fix) release helm chart as litellm-helm

ishaan-jaff committed:
(fix) releasing litellm helm as "litellm-helm"

ishaan-jaff committed:
(fix) deploy litellm-helm

ishaan-jaff committed:
(fix) deploy litellm helm

ishaan-jaff commented:
Update I took these changes and added it to main, confirmed it works. We needed to rename the chart to `litellm-helm` since it would overwrite the litellm docker image 

I'm going to still merge this PR and then re-apply my changes. Want to make sure I recognize @ShaunMaher's work on our contributors list + release notes !

Commit:
fix(caching.py): close redis connection pool upon proxy shutdown

ishaan-jaff committed:
(fix) rename chart

ishaan-jaff committed:
(fix) ghcr_deploy.yaml

ishaan-jaff committed:
(fix) ghcr_deploy.yaml

ishaan-jaff committed:
Merge branch 'update_helm_chart_deployment_2024-02-22' of https://github.com/ShaunMaher/litellm into pr/2145
